### PR TITLE
DistanceGrid : adding support of our .tmg file extension

### PR DESCRIPTION
**Pull request by thomasgosse**
This pull request aims at add a new extension file (.tmg) supported by DistanceGrid. This extension file is used to represent a shape : it's a list of values corresponding to the function evaluated at each point. All the data are placed one after the other in a single column. 
It's a minor add which does not disturb anything in SOFA, so it shouldn't raise any problems.

(edit - 08/08/17) The file is now able to provide the size of the grid : these 3 values (nx, ny, nz) are placed at the top of the .tmg file and are used at the DistanceGrid instaciation.
Example : 

**Shape.tmg**
*Header with the grid size*
>150
150
150

*Values of the grid*
> 0.015
1
1
...

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by bcarrez** [comment]
At least a little comment describing the file format in the source code would be appreciated ;-)

**Comment by guparan** [comment]
Hi Thomas,
Thank you for your first PR and welcome in the SOFA community ! :-)
Our CI is a bit messed up these days but let's try to [ci-build].

File: modules/SofaVolumetricData/DistanceGrid.cpp [hunk 1]
```diff
@@ -138,6 +138,33 @@ DistanceGrid* DistanceGrid::load(const std::string& filename,
...
+            DistanceGrid* grid = new DistanceGrid(nx, ny, nz, pmin, pmax);
+            std::ifstream inf(filename.c_str(), std::ios::in);
+            std::string content;
+            std::string::size_type sz;
+            double convert=0;
+            int i=0;
+
+            //loading in data structure
+            if(!inf) {
+                std::cout << "Unable to open .tmg file" << std::endl;
```
Comment (guparan) [inline modules/SofaVolumetricData/DistanceGrid.cpp]: Please use msg_* API for logging. `msg_error()` seems relevant here.

**Comment by thomasgosse** [comment]
@guparan It's done !

**Comment by thomasgosse** [comment]
Hi @hugtalbot, .tmg is not a standard file format, it's something that we created and which is used to store a discretization of a shape. This file format can be another way to fill a DistanceGrid (as .raw extension is) and aims at ease Thomas Morzadec's work (postdoc in DEFROST team) by allowing him to visualize his generated shapes.

**Comment by damienmarchal** [comment]
Hi all, 

It was pointed by Bruno Marques that this format if very close to mhd. So we should use it. 
As Mhd is supported by Image plugin we can close this PR.

**Pull request closed without merging**